### PR TITLE
Image segmentation and improvements

### PR DESCRIPTION
**Pull request by DanielMontesGuerrero**
- Refactor code
- Add linter
- Add tests
- Add Chan vese algorithm

### CODE REVIEW

**Review by DanielMontesGuerrero — COMMENTED** [review]
I'm a engineer, trust me bro

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+
+[dev-packages]
+
+[requires]
+python_version = "3.10"
```
Comment (EstebanOlmedo) [inline Pipfile]: Duplicated file

File: Pipfile.lock [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            {
+                "name": "pypi",
+                "url": "https://pypi.org/simple",
+                "verify_ssl": true
+            }
+        ]
+    },
+    "default": {},
+    "develop": {}
+}
```
Comment (EstebanOlmedo) [inline Pipfile.lock]: Duplicated file

File: backend/image_processing/image.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+class ImageMetadata:
+    path: str
+    name: str
+    size: int
+    has_segmented_image: bool
+    segmented: str = ''
+
+    def get_path(self, get_segmented: bool = False):
+        name = self.name if not get_segmented else self.segmented
+        return os.path.join(self.path, name)
```
Comment (EstebanOlmedo) [inline backend/image_processing/image.py:14]: Great OOP!

**Review by EstebanOlmedo — COMMENTED** [review]
Nice work! Just delete unused files, please :)

**Comment by brielgah** [comment]
I dont have more comments besides the ones that Esteban provided. Looks good! :)

**Review by brielgah — APPROVED** [review]
Looks good

**Pull request merged**